src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs

```diff
@@ -25,6 +25,19 @@ public static void AddLocationHeader(this HttpResponse response, Uri locationUrl
         response.Headers.Add(HeaderNames.Location, locationUrl.IsAbsoluteUri ? locationUrl.AbsoluteUri : GetRelativeUri(locationUrl));
     }
 
+    /// <summary>
+    /// Set Response Warning header.
+    /// </summary>
+    /// <param name="response">The httpResponse.</param>
+    /// <param name="host">Host name.</param>
+    /// <param name="message">The warning message.</param>
+    public static void SetWarning(this HttpResponse response, string host, string message)
+    {
+        EnsureArg.IsNotNull(response, nameof(response));
+        EnsureArg.IsNotEmptyOrWhiteSpace(message, nameof(message));
+        response.Headers.Warning = string.Format(DicomApiResource.WarningHeader, host ?? string.Empty, message);
```

Comment:
If `host` is `null` or empty I think it needs to be `'-'` instead

Reply:
yeah, that's right!